### DISCUSSION TITLE
Multiple video quality with encoding on source

### DISCUSSION THREAD

**Discussion by mbakumenkov** [OvenMediaLabs/OvenMediaEngine#1250] [Help] [answered]
Hello, Thank you guys for a nice software!

I`m so sorry if this question has already been asked, I couldn't find it.

I have a lot (~20) of source streams pushing via RTMP to my Origin OME -> WebRTC / LLHLS Client OvenPlayer. Streams comes in 3 qualities and in order to save origin server resources, I want to encode 3 video quality in every source I have and push it to Origin server. The problem is that in the OvenPlayer these streams looks like different sources, and when you switch to another quality, the stream reloads.

Is it possible to combine 3 streams with different stream keys into one output playlist and switch between them without reload, like if you push 1 RTMP video source and have a playlist with different encodes on the edge server?

![Screenshot from 2023-05-23 11-41-58](https://github.com/AirenSoft/OvenMediaEngine/assets/97111708/0b5a880a-4119-4d55-9d92-6cfd1ab54e2b)

**Comment by getroot** [comment, marked as answer]
Unfortunately, this feature is not provided by OME. However, you can create and provide a master playlist (m3u8 file) for the three streams. Please note that the 3 streams ingested from outside will have different PTS/DTS. Therefore, the playback position may change when switching.

**Reply by mbakumenkov** [reply]
Thanks for a quick reply!

**Comment by getroot** [comment]
It has been released! see https://github.com/AirenSoft/OvenMediaEngine/discussions/1489
